archaius2-core/src/test/java/com/netflix/archaius/config/CompositeConfigTest.java

```diff
@@ -146,6 +149,19 @@ public void getKeysTest() throws ConfigException {
         Assert.assertEquals(set("b1", "b2", "d1", "d2", "e1", "e2"), set(iter));
     }
 
+    @Test
+    public void unusedCompositeConfigIsGarbageCollected() throws ConfigException {
+        com.netflix.archaius.api.config.CompositeConfig config = DefaultCompositeConfig.builder()
```

Comment:
Did you try running this code *before* the fix? I'm not sure the bug would be triggered in this case.

Reply:
Oops, I originally had a more verbose version of these tests and I forgot to keep around the reference to the source config while trimming them down. Re-added the reference to the source config and confirmed that these tests now fail against the previous version of the code.